src/org/scada_lts/mango/service/EventService.java

```diff
@@ -66,13 +69,18 @@ public class EventService implements MangoEvent {
 	
 	private static final Log LOG = LogFactory.getLog(EventService.class);
 	
-	private final EventDAO eventDAO;
+	private final IEventDAO eventDAO;
 	private final UserEventDAO userEventDAO;
 	private final IUserCommentDAO userCommentDAO;
 	private final SystemSettingsService systemSettingsService;
 	
 	public EventService() {
-		eventDAO = new EventDAO();
+
+        if (DAO.getType() == DatabaseAccess.DatabaseType.POSTGRES) {
```

Comment:
DAO.getType() ? The getType method cannot be static, it should be instanced.

This needs to be handled in Spring according to the previously presented template from the factor class, then such if statements will not be needed.